### PR TITLE
[RLlib; docs] Docs do-over (new API stack): Env pages vol 01.

### PR DESCRIPTION
**Pull request by sven1977**
[RLlib] Docs do-over (new API stack): Env pages vol 01
* new sigils/logos for upcoming split-structure (single-,  multi-agent, external & hierarchical)
* more example scripts for simultaneous and turn-based multi-agent acting patterns (added to CI)
* remove old API stack env APIs no longer needed
* Move example multi-agent env classes into their own subfolder in `examples/envs/classes/multi_agent/..`

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/envs/classes/multi_agent/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from ray.rllib.env.multi_agent_env import make_multi_agent
```
Comment (sven1977) [inline rllib/examples/envs/classes/multi_agent/__init__.py:1]: Simply moved some example classes in here for order.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Awesome PR!

File: doc/source/rllib/images/envs/hierarchical_env_logo.svg
[the remark this answers is not in the document]
Comment (sven1977) [reply doc/source/rllib/images/envs/hierarchical_env_logo.svg:1]: Both is possible. Our example script has always only one level acting at a time.

File: rllib/examples/envs/classes/multi_agent/tic_tac_toe.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                or self.board[0:7:3] == win_val
+                or self.board[1:8:3] == win_val
+                or self.board[2:9:3] == win_val
+                # Diagonal win.
+                or self.board[::3] == win_val
+                or self.board[2:7:2] == win_val
+            ):
+                # Final reward is +5 for victory and -5 for a loss.
+                rewards[self.current_player] += 5.0
+                rewards[opponent] = -5.0
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/envs/classes/multi_agent/tic_tac_toe.py:120]: They are rewarded separately, with +1.0 and -1.0.
The misplacement penalty should be learnt pretty quickly by the agents (b/c it hurts a lot) and after that, they should be able to "focus" on the actual game, not misplacing any pieces anymore. 🤞

File: rllib/examples/envs/classes/multi_agent/tic_tac_toe.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    the opponend owns the field, and 1.0 means we occupy the field:
+    ----------
+    | 0| 1| 2|
+    ----------
+    | 3| 4| 5|
+    ----------
+    | 6| 7| 8|
+    ----------
+
+    The action space is Discrete(9) and actions landing on an alredy occupied field
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/envs/classes/multi_agent/tic_tac_toe.py:22]: done

File: rllib/examples/envs/classes/multi_agent/rock_paper_scissors.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    # __sphinx_doc_3_begin__
+    def __init__(self, config=None):
+        super().__init__()
+
+        self.agents = self.possible_agents = ["player1", "player2"]
+
+        # The observations are always the last taken actions. Hence observation- and
+        # action spaces are identical.
+        self.observation_spaces = self.action_spaces = {
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/envs/classes/multi_agent/rock_paper_scissors.py:73]: Sure, but I wanted to leave the Sheldon Cooper mode out of the docs entirely (to keep docs as simple as possible). Therefore I had to spacially separate these two logics entirely in the file.

**Pull request merged**